From d23c391658cf3bd9e19cb5e33ed45a1f475443a7 Mon Sep 17 00:00:00 2001
From: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
Date: Tue, 28 Nov 2023 19:35:04 +0530
Subject: [PATCH] soc: qcom: llcc: Add support for LLCC perfmon

LLCC perfmon will be declared as the child node of the LLCC node. So,
add support to populate it in the LLCC probe.

Change-Id: Ia283e38f1415ee2b2e5d21d8bf35c7bf52b79c8f
Signed-off-by: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
---
 drivers/soc/qcom/llcc-qcom.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/soc/qcom/llcc-qcom.c b/drivers/soc/qcom/llcc-qcom.c
index 5c34c58d4d8c..6cf8c1c9f888 100644
--- a/drivers/soc/qcom/llcc-qcom.c
+++ b/drivers/soc/qcom/llcc-qcom.c
@@ -17,6 +17,7 @@
 #include <linux/sizes.h>
 #include <linux/slab.h>
 #include <linux/soc/qcom/llcc-qcom.h>
+#include <linux/of_platform.h>
 
 #define ACTIVATE                      BIT(0)
 #define DEACTIVATE                    BIT(1)
@@ -1068,6 +1069,9 @@ static int qcom_llcc_probe(struct platform_device *pdev)
 			dev_err(dev, "Failed to register llcc edac driver\n");
 	}
 
+	if (of_platform_populate(dev->of_node, NULL, NULL, dev))
+		dev_err(dev, "llcc perfmon populate failed!!\n");
+
 	return 0;
 err:
 	drv_data = ERR_PTR(-ENODEV);
-- 
2.34.1

